src/game/Spell.cpp

```diff
@@ -4508,6 +4508,14 @@ SpellCastResult Spell::CheckCast(bool strict)
         {
             case SPELL_EFFECT_DUMMY:
             {
+                //By Spell ID
+                if (m_spellInfo->Id == 19938)               // Awaken Lazy Peon
+                {
+                    Unit* target = m_targets.getUnitTarget();
+                    if (!target || !target->HasAura(17743 /*SPELL_PEON_SLEEP*/) || target->GetEntry() != 10556 /*NPC_SLEEPING_PEON*/)
+                        return SPELL_FAILED_BAD_TARGETS;
+                }
+                //By SpellIconID
```

Comment:
should make the line below 'else if' instead of 'if' - no reason to execute the checks if we just ran the if above.